src/varnam_state.cpp

```diff
@@ -370,11 +370,15 @@ void VarnamState::selectCandidate(int index) {
 void VarnamState::commitText(const FcitxKeySym &key) {
   auto candidates = ic_->inputPanel().candidateList();
   std::string stringToCommit;
+  bool isWordBreakKey = isWordBreak(key);
 
-  if (key == FcitxKey_Escape || key == FcitxKey_0 || candidates == nullptr ||
-      candidates->size() <= 1 || result_ == nullptr ||
-      varray_is_empty(result_) || lastTypedCharIsDigit) {
-
+if (key == FcitxKey_space) {
```

Comment:
Could you give your insights on why this block is added. And why do we have to handle space key specifically?.
A better approach would be to use `enableIndicNumbers`config  with `lastTypedCharIsDigit` I think.

Reply:
numbers were shown in suggestion but they are not stringToCommit. i was not able to commit numbers without this change.